rio_tiler/mosaic/reader.py

```diff
@@ -73,7 +73,7 @@ def reader(asset: str, *args, **kwargs) -> Tuple[numpy.ndarray, numpy.ndarray]:
         )
 
     if not chunk_size:
-        chunk_size = threads or len(assets)
+        chunk_size = threads if threads > 1 else len(assets)
```

Comment:
this explains why it was fine with `threads=1`, because we were processing chunks of one element.